lib/about_page/configuration.rb

```diff
@@ -63,21 +64,35 @@ def set_headers! response
     end
 
     def to_h
-      self.hash.to_h.inject({}) { |h,v| h[v[0]] = v[1].respond_to?(:to_h) ? v[1].to_h : v[1]; h }
+      self.hash.to_h.inject({}) { |h, v| h[v[0]] = v[1].respond_to?(:to_h) ? v[1].to_h : v[1]; h }
     end
 
     def health_report
-      self.nodes.collect do |key, profile| 
-        if profile.class.validators.length > 0 
-          health = profile.valid? ? 'ok' : 'error'
-          errors = profile.errors.map {|a,m| "#{a} #{m}"}
-          { 'component' => key.to_s, 'status' => health, 'errors' => errors }
-        else
-          nil
+      self.nodes.collect do |key, profile|
+        next unless profile.class.validators.length.positive?
+
+        begin
+          health = Timeout.timeout(55) { profile.valid? ? 'ok' : 'error' }
```

Comment:
If one component times out then the rest of the components need to run in <5 sec in order for nagios not to give a general warning.  I wonder if we should allow the configuration to specify a timeout threshold and fall back to something medium sized (20?) so if multiple services (solr and fedora?) are being slow it hopefully won't nagios timeout.  With a threshold in configuration then we can tune the specific thresholds for each component (30s Fedora, 20s Solr, 5s for each other component or something like that).  What do you think?